src/engine/strat_engine/engine.rs

```diff
@@ -52,8 +53,10 @@ impl StratEngine {
     ///
     /// Returns an error if the kernel doesn't support required DM features.
     /// Returns an error if there was an error reading device nodes.
+    /// Returns an error if the binaries on which it depends can not be found.
     pub fn initialize() -> StratisResult<StratEngine> {
         let dm = get_dm_init()?;
+        verify_binaries()?;
```

Comment:
If for some reason a user has an existing stratis setup, they update their system and during the update they mess up the required binaries the stratisd daemon will exit with error.  If we aren't running we cannot monitor the thin pool(s) for out of space and other errors.  Having systemd unit file instructing us to restart us isn't going to help as we will just exit until systemd gives up.  I'm wondering if not exiting here would be better?  This is much better than my proposed PR where we just panic.

Reply:
If they just moved the binaries to a place where we expect, stratisd will find them again and all will be well. I don't see why this should happen, but who knows?

If some of the binaries that we rely on are actually gone, then stratisd can not function properly, anymore than it could function properly if devicemapper context is missing. We might as well exit.

In a future, we could work to modify stratisd so that it runs in some kind of crippled mode. But that would be complex. Maybe the future that comes sooner is that all or most of these become available as libraries! That would be marvelous.